pages/_app.tsx

```
@@ -1,11 +1,21 @@
-import App from 'next/app';
-import React from 'react';
+import type { NextPage } from 'next';
+import type { AppProps } from 'next/app';
+import type { ReactElement, ReactNode } from 'react';
 
-class MyApp extends App {
-  render() {
-    const { Component, pageProps } = this.props;
+export type NextPageWithLayout<P = {}, IP = P> = NextPage<P, IP> & {
+  getLayout?: (page: ReactElement, isBrowserOutdated: boolean) => ReactNode;
+};
 
-    return <Component {...pageProps} />;
-  }
+type AppPropsWithLayout = AppProps & {
+  Component: NextPageWithLayout;
+};
+
+export default function MyApp({ Component, pageProps }: AppPropsWithLayout) {
+  // Use the layout defined at the page level, if available
```

What happens when no layout defined ? Should we raise an exception ?